pnpm@7.7.1 - pnpm generates different lockfile locally and on CI

<!--
  If this issue affects many people in a company/big team, create a post for your company in the following discussion:
  https://github.com/pnpm/pnpm/discussions/3787
  and link the issue in your post.

  This will help us prioritize issues that affect more people.
-->

### pnpm version:

7.7.0
7.7.1

### Code to reproduce the issue:

https://github.com/gionkunz/chartist-js

### Expected behavior:

successful dependencies installation 

### Actual behavior:

```
Lockfile is up-to-date, resolution step is skipped
 ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up-to-date with package.json

Note that in CI environments this setting is true by default. If you still need to run install in such cases, use "pnpm install --no-frozen-lockfile"
```

### Additional information:

[v7.5.0 works perfect](https://github.com/gionkunz/chartist-js/runs/7590723454?check_suite_focus=true)

[v7.7.1 installed local and on CI, locally lockfile was updated and pushed, but on CI it still shows that lockfile is not up-to-date with package.json](https://github.com/gionkunz/chartist-js/runs/7590711885?check_suite_focus=true#step:5:11).

--- a/.changeset/eighty-cheetahs-shout.md
+++ b/.changeset/eighty-cheetahs-shout.md
@@ -1,0 +1,6 @@
+---
+"@pnpm/lockfile-file": patch
+"pnpm": patch
+---
+
+Don't incorrectly identify a lockfile out-of-date when the package has a publishConfig.directory field [#5124](https://github.com/pnpm/pnpm/issues/5124).

--- a/packages/lockfile-file/src/read.ts
+++ b/packages/lockfile-file/src/read.ts
@@ -231,6 +231,7 @@ function convertFromLockfileFileMutable (lockfileFile: LockfileFile): Lockfile {
       '.': {
         specifiers: lockfileFile['specifiers'] ?? {},
         dependenciesMeta: lockfileFile['dependenciesMeta'],
+        publishDirectory: lockfileFile['publishDirectory'],
       },
     }
     delete lockfileFile.specifiers

--- a/packages/lockfile-file/test/read.test.ts
+++ b/packages/lockfile-file/test/read.test.ts
@@ -27,6 +27,7 @@ test('readWantedLockfile()', async () => {
         dependenciesMeta: {
           foo: { injected: true },
         },
+        publishDirectory: undefined,
       },
     })
   }

